kubernetes/CMakeLists.txt

```diff
@@ -1546,14 +1548,66 @@ include(PreTarget.cmake OPTIONAL)
 add_library(${pkgName} ${SRCS} ${HDRS})
 # Link dependent libraries
 if(NOT CMAKE_VERSION VERSION_LESS 3.4)
-    target_link_libraries(${pkgName} OpenSSL::SSL OpenSSL::Crypto)
+    target_link_libraries(${pkgName} PRIVATE OpenSSL::SSL OpenSSL::Crypto)
 endif()
-target_link_libraries(${pkgName} ${CURL_LIBRARIES} )
+target_link_libraries(${pkgName} PUBLIC ${CURL_LIBRARIES} )
+target_include_directories(
+    ${pkgName} PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
+    $<INSTALL_INTERFACE:include>
+)
 
 include(PostTarget.cmake OPTIONAL)
 
-#install library to destination
-install(TARGETS ${pkgName} DESTINATION ${CMAKE_INSTALL_PREFIX})
+# installation of libraries, headers, and config files
+if (NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/Config.cmake.in)
+    install(TARGETS ${pkgName} DESTINATION ${CMAKE_INSTALL_PREFIX})
+else()
+    include(GNUInstallDirs)
+    install(TARGETS ${pkgName}
```

Comment:
`cmake ..` will fail in my env:
```
CMake Error at CMakeLists.txt:1566 (install):
  install TARGETS given no LIBRARY DESTINATION for shared library target
  "kubernetes".


CMake Error at CMakeLists.txt:1584 (export):
  export Export set "kubernetesTargets" not found.
```

After I changed the line 1655
to
```cmake
install(TARGETS ${pkgName} DESTINATION ${CMAKE_INSTALL_PREFIX}
```
It works.

Reply:
I can build and install on both Ubuntu 20.04 and Windows without any problem.

Reply:
I know. But on my env (Ubuntu 18.04 and default cmake), this change is needed. I think "DESTINATION" is required but new version of cmake can ignore it.

Reply:
Requested changes made and a new PR opened https://github.com/OpenAPITools/openapi-generator/pull/10378